logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/WaitUntilLiveUseCase.kt

```diff
@@ -0,0 +1,17 @@
+package com.wire.kalium.logic.sync
+
+import com.wire.kalium.logic.data.sync.SyncState
+
+/**
+ * Starts sync (if not yet started) and suspends until all pending events were processed.
+ * Will resume when all pending events are processed and the client is considered "online".
+ * @see SyncState
+ */
+class WaitUntilLiveUseCase(
```

Comment:
Is this going to be used in reloaded?  If that's the case, maybe we can extract an interface out of this and make the implementation internal. What do you think ? 😬

Reply:
The idea was that, to expose a way to suspend until the app is online and only then process the notifications.

But perhaps we can instead adapt the UseCases to do it, and Reloaded won't need to call this.

I kinda prefere this second way. I'll adapt this PR to only remove the deprecated UseCases, and we can figure out the `waitUntilLive` later.